components/movementsensor/gpsnmea/pmtkI2C.go

```diff
@@ -186,7 +192,7 @@ func (g *PmtkI2CNMEAMovementSensor) Start(ctx context.Context) error {
 						}
 					}
 					strBuf = ""
-				} else if b != 0x0A && b != 0xFF { // adds only valid bytes
+				} else if b != 0x0A && b < 0x7F { // adds only valid bytes
```

Comment:
[optional] The comment is about only adding valid bytes. Taking that a step further: the only printable ASCII characters are in the range 0x20 through 0x7E. I wonder if the first part should be `b >= 0x20` instead of just `b != 0x0A`. That might be overkill: maybe it's fine as-is.